### ISSUE TITLE
[2.0] "'500 Internal Server Error' caused by: all servers have 0 weight"

### ISSUE THREAD

**Issue by andig** [containous/traefik#4828]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

Declare single service (see toml below) then

    ❯ curl -H "Host:volkszaehler.io" localhost:8080
    Internal Server Error

and the traefik log shows:

    ❯ docker run -v /var/run/docker.sock:/var/run/docker.sock -v /Volumes/data/docker:/etc -p 8080:8080 traefik:v2.0 -c /etc/traefik/traefik2.toml --global.sendanonymoususage 0 --log.level=debug
    time="2019-04-28T12:39:00Z" level=info msg="Using TOML configuration file /etc/traefik/traefik2.toml"
    time="2019-04-28T12:39:00Z" level=info msg="Traefik version 2.0.0-alpha4 built on 2019-04-17T11:38:21Z"
    time="2019-04-28T12:39:00Z" level=debug msg="Static configuration loaded {\"Global\":{\"Debug\":false,\"CheckNewVersion\":true,\"SendAnonymousUsage\":false},\"ServersTransport\":{\"InsecureSkipVerify\":false,\"RootCAs\":null,\"MaxIdleConnsPerHost\":200,\"ForwardingTimeouts\":null},\"EntryPoints\":{\"http\":{\"Address\":\":8080\",\"Transport\":{\"LifeCycle\":{\"RequestAcceptGraceTimeout\":0,\"GraceTimeOut\":10000000000},\"RespondingTimeouts\":{\"ReadTimeout\":0,\"WriteTimeout\":0,\"IdleTimeout\":180000000000}},\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":false,\"TrustedIPs\":null}}},\"Providers\":{\"ProvidersThrottleDuration\":0,\"Docker\":null,\"File\":{\"Directory\":\"\",\"Watch\":true,\"Filename\":\"\",\"DebugLogGeneratedTemplate\":false,\"TraefikFile\":\"/etc/traefik/traefik2.toml\"},\"Marathon\":null,\"Kubernetes\":null,\"KubernetesCRD\":null,\"Rest\":null,\"Rancher\":null},\"API\":null,\"Metrics\":null,\"Ping\":null,\"Log\":{\"Level\":\"debug\",\"format\":\"common\"},\"AccessLog\":null,\"Tracing\":null,\"HostResolver\":null,\"ACME\":null}"
    time="2019-04-28T12:39:00Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n"
    time="2019-04-28T12:39:00Z" level=debug msg="No default certificate, generate one"
    time="2019-04-28T12:39:00Z" level=info msg="Starting provider aggregator.ProviderAggregator {}"
    time="2019-04-28T12:39:00Z" level=debug msg="Start TCP Server" entryPointName=http
    time="2019-04-28T12:39:00Z" level=info msg="Starting provider *file.Provider {\"Directory\":\"\",\"Watch\":true,\"Filename\":\"\",\"DebugLogGeneratedTemplate\":false,\"TraefikFile\":\"/etc/traefik/traefik2.toml\"}"
    time="2019-04-28T12:39:00Z" level=debug msg="Configuration received from provider file: {\"HTTP\":{\"routers\":{\"vz\":{\"entryPoints\":null,\"service\":\"vz\",\"rule\":\"Host(`domain.io`)\"}},\"services\":{\"vz\":{\"loadbalancer\":{\"servers\":[{\"url\":\"http://server.fritz.box:8080\",\"weight\":0}],\"passHostHeader\":false}}}},\"TCP\":{},\"TLSOptions\":{},\"TLSStores\":{}}" providerName=file
    time="2019-04-28T12:39:00Z" level=debug msg="Creating middleware" entryPointName=http routerName=file.vz serviceName=vz middlewareName=pipelining middlewareType=Pipelining
    time="2019-04-28T12:39:00Z" level=warning msg="Invalid load-balancing method \"\", fallback to 'wrr' method" entryPointName=http routerName=file.vz serviceName=vz
    time="2019-04-28T12:39:00Z" level=debug msg="Creating wrr load-balancer" entryPointName=http routerName=file.vz serviceName=vz
    time="2019-04-28T12:39:00Z" level=debug msg="Creating server 0 at http://server.fritz.box:8080 with weight 0" serviceName=vz entryPointName=http routerName=file.vz serverName=0
    time="2019-04-28T12:39:00Z" level=debug msg="Added outgoing tracing middleware vz" middlewareType=TracingForwarder entryPointName=http routerName=file.vz middlewareName=tracing
    time="2019-04-28T12:39:00Z" level=debug msg="Creating middleware" entryPointName=http middlewareName=traefik-internal-recovery middlewareType=Recovery
    time="2019-04-28T12:39:00Z" level=debug msg="No default certificate, generate one"
    time="2019-04-28T12:39:05Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"*/*\"],\"User-Agent\":[\"curl/7.54.0\"],\"X-Forwarded-Host\":[\"domain.io\"],\"X-Forwarded-Port\":[\"80\"],\"X-Forwarded-Proto\":[\"http\"],\"X-Forwarded-Server\":[\"94322a3f8e2b\"],\"X-Real-Ip\":[\"172.17.0.1\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"domain.io\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"172.17.0.1:60358\",\"RequestURI\":\"/\",\"TLS\":null}"
    time="2019-04-28T12:39:05Z" level=debug msg="'500 Internal Server Error' caused by: all servers have 0 weight"
    time="2019-04-28T12:39:05Z" level=debug msg="vulcand/oxy/roundrobin/rr: completed ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/1.1\",\"ProtoMajor\":1,\"ProtoMinor\":1,\"Header\":{\"Accept\":[\"*/*\"],\"User-Agent\":[\"curl/7.54.0\"],\"X-Forwarded-Host\":[\"domain.io\"],\"X-Forwarded-Port\":[\"80\"],\"X-Forwarded-Proto\":[\"http\"],\"X-Forwarded-Server\":[\"94322a3f8e2b\"],\"X-Real-Ip\":[\"172.17.0.1\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"domain.io\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"172.17.0.1:60358\",\"RequestURI\":\"/\",\"TLS\":null}"

### What did you expect to see?

Request handled for single server

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.0.0-alpha4
Codename:     faisselle
Go version:   go1.12.4
Built:        2019-04-17T11:38:21Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

    ❯ docker run -v /var/run/docker.sock:/var/run/docker.sock -v /Volumes/data/docker:/etc -p 8080:8080 traefik:v2.0 -c /etc/traefik/traefik2.toml --global.sendanonymoususage 0 --log.level=debug

```toml
[entryPoints]
  [entryPoints.http]
    address = ":8080"

[providers]
  [providers.file]

[http]
  [http.routers]
    [http.routers.vz]
      rule = "Host(`domain.io`)"
      service = "vz"

  [http.services]
    [http.services.vz]
      [[http.services.vz.LoadBalancer.servers]]
        url = "http://server.fritz.box:8080"
```

**Comment by andig** [comment]
Seems I have to add `weight = 1` for a single server. Would be great if that would be the default value for such a simple case?

**Comment by ldez** [comment]
Hello,

It's the expected behavior with the v2.

When you are using the file provider you have to define explicitly the weight to 1.

**Comment by polarathene** [comment]
@ldez I have just spent hours looking into why I couldn't get the file provider to properly work while I was able to connect to the service via other means.

There is no indication that this is required at all, nothing shows up in logs for me, all I would get is `Internal Server Error`. [v2 docs have an example for a single server with file provider](https://docs.traefik.io/v2.0/routing/services/) and there is **no** usage of a `weight = 1` anywhere there :sob: 

> It's the expected behavior with the v2.

How are users meant to become aware of this? I'll raise a new issue about the docs. Are there more gotchas like this?
